glide-core/redis-rs/redis/src/cluster_routing.rs

```diff
@@ -1336,18 +1566,95 @@ mod tests {
         let res3 = Value::Array(vec![Value::SimpleString("2".to_string()), Value::Int(3)]);
         let results = super::combine_and_sort_array_results(
             vec![res1, res2, res3],
-            [vec![0, 5], vec![1, 4], vec![2, 3]].iter(),
+            &[
+                (Route(4813, SlotAddr::Master), vec![2, 3]),
+                (Route(5061, SlotAddr::Master), vec![1, 4]),
+                (Route(12182, SlotAddr::Master), vec![0, 5]),
+            ],
+            &MultiSlotArgPattern::KeysOnly,
         );
 
         assert_eq!(
             results.unwrap(),
             Value::Array(vec![
-                Value::Nil,
-                Value::BulkString("1".as_bytes().to_vec()),
                 Value::SimpleString("2".to_string()),
-                Value::Int(3),
+                Value::BulkString("1".as_bytes().to_vec()),
+                Value::Nil,
+                Value::Okay,
                 Value::BulkString("4".as_bytes().to_vec()),
+                Value::Int(3),
+            ])
+        );
+    }
+
+    #[test]
+    fn test_combining_results_into_single_array_key_value_paires() {
+        // For example `MSET foo bar foo2 bar2 {foo}foo3 bar3`
+        let res1 = Value::Array(vec![Value::Okay]);
+        let res2 = Value::Array(vec![Value::BulkString("1".as_bytes().to_vec()), Value::Nil]);
+        let results = super::combine_and_sort_array_results(
+            vec![res1, res2],
+            &[
+                (Route(1044, SlotAddr::Master), vec![2, 3]),
+                (Route(12182, SlotAddr::Master), vec![0, 1, 4, 5]),
+            ],
+            &MultiSlotArgPattern::KeyValuePairs,
+        );
+
+        assert_eq!(
+            results.unwrap(),
+            Value::Array(vec![
+                Value::BulkString("1".as_bytes().to_vec()),
+                Value::Okay,
+                Value::Nil
+            ])
+        );
+    }
+
+    #[test]
+    fn test_combining_results_into_single_array_keys_and_path() {
+        // For example `JSON.MGET foo bar {foo}foo2 $.a`
+        let res1 = Value::Array(vec![Value::Okay]);
+        let res2 = Value::Array(vec![Value::BulkString("1".as_bytes().to_vec()), Value::Nil]);
+        let results = super::combine_and_sort_array_results(
+            vec![res1, res2],
+            &[
+                (Route(5061, SlotAddr::Master), vec![2, 3]),
+                (Route(12182, SlotAddr::Master), vec![0, 1, 3]),
+            ],
+            &MultiSlotArgPattern::KeysAndPath,
+        );
+
+        assert_eq!(
+            results.unwrap(),
+            Value::Array(vec![
+                Value::BulkString("1".as_bytes().to_vec()),
+                Value::Nil,
+                Value::Okay,
+            ])
+        );
+    }
+
+    #[test]
+    fn test_combining_results_into_single_array_key_path_value_triples() {
```

Comment:
What is the expected behavior of cases where 1 node succeeded to execute the MSET but the other instance failed?

Reply:
The response would be an error. Added more detailed description to the relevant multi slot APIs in https://github.com/valkey-io/valkey-glide/commit/b25442457f1ef7faa5256529ae3452cc8c2d4be5